crates/pathfinder/src/state/sync.rs

```diff
@@ -158,17 +158,25 @@ where
         let l2_head = tx
             .block_header(pathfinder_storage::BlockId::Latest)
             .context("Fetching latest block header from database")?
-            .map(|header| (header.number, header.hash, header.state_commitment));
+            .map(|header| {
+                (
+                    header.number,
+                    header.hash,
+                    header.state_commitment,
+                    header.starknet_version,
+                )
+            });
```

Comment:
Probably worth just making this `BlockHeader`. Back when I did this nonsense we didn't have the header type available yet.

Reply:
Unfortunately that's not really easy to do. The commitment fields in `BlockHeader` are not in a block as fetched from the feeder gateway, so we can't easily produce a `BlockHeader` in all places we're using this tuple right now.

I'm sure we could improve the current code but it's far from trivial and would require extensive changes to L2 sync / reorg code which I'm hesitant to make within this PR -- especially given that this whole thingy here is temporary. We can revert the `starknet_version` related changes as soon as 0.12.2 has been rolled out to mainnet...

Reply:
Ah yeah we would have to start filling them in. Will become a lot easier once we switch to the combined end point for normal sync since then we can actually calculate / copy these over.